src/sources/aws_kinesis_firehose/mod.rs

```diff
@@ -295,13 +300,16 @@ mod tests {
                 record.to_owned(),
                 record.to_owned(),
             ),
-            (
-                Compression::Gzip,
-                Compression::Gzip,
-                true,
-                Vec::new(),
-                Vec::new(),
-            ),
+            // TODO: This test case doesn't work, because currently `codecs::BytesCodec` does not
+            // support forwarding empty frames. This should be addressed and this test case
+            // reenabled.
```

Comment:
Should we open an issue for this?

Reply:
This would be a regression, I don't think we can ship this until we address it.

Ref: https://github.com/vectordotdev/vector/issues/7710

Reply:
@jszwedko that issue is very related, but I don't think this change would regress to that error. It would fail to produce any event, so there would be no attempt to decode an empty file. This should still be addressed of course and the test case was a good canary of surfacing the bug in `codecs::BytesCodec`.

Opened #9398 for that. After #9399 lands, those changes should also be gone from the diff in this PR.